### PR TITLE
Add more leak retention fixes

### PR DESCRIPTION
**Pull request by Therzok**
This brings down all ProjectFile and Document leaks to 0 on the stress test app.

### CODE REVIEW

**Comment by Therzok** [comment]
Nice, this uncovered use-after-dispose issues.

**Comment by Therzok** [comment]
Something is happening on CI making the IDE not have focus, thus failing the test. Rebasing.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopWorkspace.ProjectSystemHandler.cs [hunk 1]
```diff
@@ -266,6 +266,9 @@ async Task<SolutionInfo> CreateSolutionInfoInternal (MonoDevelop.Projects.Soluti
 							return await CreateSolutionInfoInternal (solution, token).ConfigureAwait (false);
 						}
 
+						if (token.IsCancellationRequested)
+							return (solution, null);
```
Comment (alanmcgovern) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopWorkspace.ProjectSystemHandler.cs]: why not `token.ThrowIfCancellationRequested ()`?

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopWorkspace.ProjectSystemHandler.cs]: I don't want to bring down the entire IDE :p

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopWorkspace.ProjectSystemHandler.cs]: This exception will not bubble to GetWorkspaceAsync, but this is the internal workspace loading mechanism and we don't want it to throw.

So return null here and handle that to pass on to the actual tcs which will set it's been canceled.

**Comment by Therzok** [comment]
Closing for now. Will add pieces of this PR in other PRs.

**Pull request closed without merging**